### PR TITLE
DOCUMENTATION: Add developer documentation

### PR DESCRIPTION
**Pull request by Vivelis**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [ ] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [ ] Feature request
- [x] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**
Add developer documentation

* **Issue ticket number and link**
[AR-90](https://bravmm.atlassian.net/browse/AR-90)

[AR-90]: https://bravmm.atlassian.net/browse/AR-90?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- **Documentation**
  - Introduced comprehensive developer documentation covering setup and usage for backend, database, frontend, mobile, and services API.
  - Launched user documentation detailing project overview, account creation, login procedures, service connection, and automation creation.

### CODE REVIEW

File: docs/developer_documentation/back/index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+CRYPTO_SECRET="crypto_secret"
+# Crypto algorithm
+CRYPTO_ALGO="aes-..."
+# Crypto iv
+SALT_ROUNDS=10
+# JWT token expiration
+JWT_TOKEN_EXPIRATION="1h"
+# Interval between each service check
+INTERVAL=60000
+```
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/developer_documentation/back/index.md:50]: it's ok

File: docs/developer_documentation/index.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+├── front
+│   ├── Dockerfile
+│   ├── package.json
+│   ├── .env.development
+│   ├── app/
+├── mobile
+│   ├── Dockerfile
+│   ├── package.json
+│   ├── app/
+```
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/developer_documentation/index.md:84]: it's ok

File: docs/developer_documentation/db/index.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```bash
+docker build -t database .
+```
+
+Then, you can run the container:
+
+```bash
+docker run -d -p 5432:5432 database
+```
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/developer_documentation/db/index.md:40]: it's ok

File: docs/developer_documentation/mobile/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+Scan the QR code displayed in the terminal or copy the link displayed in the terminal and paste it in the Expo Go application.
+It will automatically open the application.
+
+### Using an emulator
+
+You can use an emulator to run the application on your computer.
+To do this, you need to install the [Android Studio](https://developer.android.com/studio) or [Xcode](https://developer.apple.com/xcode/) IDE.
+Then, you need to create a virtual device.
+
+Finally, pres `a` in the terminal to run the application on the Android emulator or `i` to run the application on the iOS emulator.
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/developer_documentation/mobile/index.md:57]: It's ok

File: docs/user_documentation/index.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+To log in to the project, you must first have created an account (see [User creation](#user-creation)).
+Then, you can log in to the project.
+
+1. Go to the [login page](http://localhost:3000/login)
+2. Fill in the form
+3. Click on the "Login" button
+
+### Connect a service
+
+### Create an automation
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/user_documentation/index.md:44]: it's ok

File: docs/developer_documentation/services_api/index.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+- [Discord](Discord.md)
+- [GoogleDrive](GoogleDrive.md)
+- [GoogleMail](GoogleMail.md)
+- [MicrosoftDrive](MicrosoftDrive.md)
+- [MicrosoftMail](MicrosoftMail.md)
+- [Spotify](Spotify.md)
+- [Teams](Teams.md)
+- [Tracker.gg](Tracker.gg.md)
+- [Twitch](Twitch.md)
+- [WeatherApi](WeatherApi.md)
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/developer_documentation/services_api/index.md:17]: It's done in the linked doc

File: docs/developer_documentation/back/index.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+```bash
+docker build -t a-rien/back .
+```
+
+Then, you can run the Docker image.
+
+```bash
+docker run -d -p 8080:8080 --env-file .env a-rien/back
+```
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply docs/developer_documentation/back/index.md:73]: it's ok

**Review by romainpanno — APPROVED** [review]

**Review by Beafowl-Pull — APPROVED** [review]
RAS

**Pull request merged**